lib/transaction.js

```diff
@@ -91,6 +103,56 @@ function isModelClass(maybeModel) {
   return isFunction(maybeModel) && maybeModel.isObjectionModelClass;
 }
 
+async function waitForRunAfterTransaction(
+  initialIndex = 0,
+  runAfterTransactionMethods,
+  afterTransactionMethodsBatch,
+  failed = false,
+) {
+  if (runAfterTransactionMethods.length > 0) {
+    const promises = [];
+
+    for (let index = initialIndex; index < runAfterTransactionMethods.length; index++) {
+      promises.push(
+        promiseUtils.try(() => {
+          return runAfterTransactionMethods[index]();
+        }),
+      );
+      if (index === runAfterTransactionMethods.length - 1) {
+        return Promise.all(promises)
+          .then(() => {
+            if (failed) {
+              return Promise.reject(new Error('Run After transaction failed.'));
+            }
+          })
+          .catch(() => {
+            return Promise.reject(new Error('Run After transaction failed.'));
+          });
+      }
+      if ((index + 1) % afterTransactionMethodsBatch === 0) {
```

Comment:
as there could be many methods to be executed after transaction, better to do this operation in batches. In order to prevent thousands of promises in parallel that consume more resources that are available